lib/flight_scheduler/allocation_registry.rb

```diff
@@ -244,4 +240,37 @@ def clear
       @node_allocations.clear
     end
   end
+
+  def considering_allocations(node, given_allocations, excluding_jobs, reservations)
+    allocations = nil
+    if given_allocations
+      allocations = given_allocations
```

Comment:
There should be a filter here to ensure the `node` is part of the `given_allocations`. This will be an easy mistake to make and something similar is already done for `reservations`.

Reply:
Good catch.